packages/jupyter-ai/src/components/chat-messages.tsx

```diff
@@ -106,24 +105,26 @@ export function ChatMessages(props: ChatMessagesProps) {
   const [timestamps, setTimestamps] = useState<Record<string, string>>({});
 
   /**
-   * Effect: update cached timestamp strings upon receiving a new message and
-   * every 5 seconds after that.
+   * Effect: update cached timestamp strings upon receiving a new message.
    */
   useEffect(() => {
     function updateTimestamps() {
       const newTimestamps: Record<string, string> = {};
       for (const message of props.messages) {
-        newTimestamps[message.id] =
-          formatDistanceToNowStrict(fromUnixTime(message.time)) + ' ago';
+        if (!(message.id in newTimestamps)) {
+          // Use the browser's default locale
+          newTimestamps[message.id] =
+            new Date(message.time * 1000) // Convert message time to milliseconds
+              .toLocaleTimeString([], {
+                hour: 'numeric', // Avoid leading zero for hours; we don't want "03:15 PM"
+                minute: '2-digit'
+              });
+        }
```

Comment:
Should `timestamps` state variable be updated only when new timestamp is generated vs every `updateTimestamps()` call to avoid unnecessary re-renders? Admittedly scenario where reference to `props.messages` changes without messages themselves changing is unlikely with how messages are currently fetched. But change is small and would make `ChatMessages` logic better.

Reply:
Updated